test/test_utf8.cpp

```diff
@@ -218,12 +249,36 @@ TEST(Compare_Core_utf8_invalid_crash)
     }
 }
 
-
 TEST(UTF8_Compare_Core_utf8_zero)
 {
-    auto str_compare = [](StringData a, StringData b) {
-        return a < b;
+    Array parent(Allocator::get_default());
+    parent.create(NodeHeader::type_HasRefs, false, 1, 0);
+    StringInterner interner(Allocator::get_default(), parent, ColKey(0), true);
+
+    // intern all the utf8 strings
+    const char* empty = "\0";
+    const char* a = "a";
+    const char* aa = "a\0a";
+    const char* ab = "a\0b";
+    const char* a0 = "a\0";
+    const char* a00 = "a\0\0";
+    interner.intern(StringData(empty, 1));
+    interner.intern(StringData(a));
+    interner.intern(StringData(aa, 3));
+    interner.intern(StringData(ab, 3));
+    interner.intern(StringData(a0, 2));
+    interner.intern(StringData(a00, 3));
+
+    auto str_compare = [&interner, this](StringData a, StringData b) {
+        auto id1 = interner.lookup(a);
+        auto id2 = interner.lookup(b);
```

Comment:
You could intern the strings on the fly, if they don't exist already. Then you don't need to worry about adding them manually. That should reduce the code duplication and make it easier to add more strings in the future.

With this pattern, you should be able to easily adapt the `UTF8_Compare_Strings` test as well.